CLI: Clean --deep code and add --pro and --fast-deep

IagoAbal opened the pull request:
Closes PA-2440

test plan:
```
% semgrep --pro -l apex -e 'foo(...)' \
      semgrep-proprietary/tests/patterns/apex/dots_string.cls
% semgrep --fast-deep \
      -c semgrep-proprietary/tests/tainting/python/simple_intrafile_eval/deep.yaml \
      semgrep-proprietary/tests/tainting/python/simple_intrafile_eval/intrafile.py
```

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

aryx commented on changelog.d/pa-2440.changed:
```diff
@@ -0,0 +1,3 @@
+PRO: Add **experimental** flags --pro and --fast-deep. Using --pro you can enable
+Apex support, and with --fast-deep you can enable intra-file inter-procedural
+taint analysis.
```
maybe you could add '(both require to first run semgrep install proxxx)'

aryx commented on cli/src/semgrep/core_runner.py:
```diff
@@ -518,14 +519,16 @@ class CoreRunner:
     ):
-        self._jobs = jobs if jobs else 1 if deep else get_cpu_count()
+        self._jobs = (
+            jobs if jobs else 1 if engine is EngineType.DEEP_INTER else get_cpu_count()
```
not a huge fan of this postfix ifthenelse.

IagoAbal replied:
I agree that in this case, with two of those nested, it is more clear to use a separate if-statement.

IagoAbal committed:
CLI: Clean --deep code and add --pro and --fast-deep

Closes PA-2440

test plan:
% semgrep --pro -l apex -e 'foo(...)' \
      semgrep-proprietary/tests/patterns/apex/dots_string.cls
% semgrep --fast-deep \
      -c semgrep-proprietary/tests/tainting/python/simple_intrafile_eval/deep.yaml \
      semgrep-proprietary/tests/tainting/python/simple_intrafile_eval/intrafile.py

IagoAbal committed:
Yoann's review
